components/tidb_query/src/codec/mysql/binary_literal.rs

```diff
@@ -0,0 +1,466 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cmp::{Eq, Ord, Ordering, PartialEq, PartialOrd};
+use std::string::ToString;
+
+use byteorder::{BigEndian, WriteBytesExt};
+use hex;
+
+use crate::codec::error::Error;
+use crate::codec::Result;
+use crate::expr::EvalContext;
+
+/// BinaryLiteral is the internal type for storing bit / hex literal type.
+#[derive(Debug)]
+pub struct BinaryLiteral(Vec<u8>);
+
+fn trim_leading_zero_bytes(bytes: &[u8]) -> &[u8] {
+    if bytes.is_empty() {
+        return bytes;
+    }
+    let (pos, _) = bytes
+        .iter()
+        .enumerate()
+        .find(|(_, &x)| x != 0)
+        .unwrap_or((bytes.len() - 1, &0));
+    &bytes[pos..]
+}
+
+/// Returns the int value for the literal.
+pub fn to_uint(ctx: &mut EvalContext, bytes: &[u8]) -> Result<u64> {
+    let bytes = trim_leading_zero_bytes(bytes);
+    if bytes.is_empty() {
+        return Ok(0);
+    }
+    if bytes.len() > 8 {
+        ctx.handle_truncate_err(Error::truncated_wrong_val(
+            "BINARY",
+            BinaryLiteral(bytes.to_owned()).to_string(),
+        ))?;
+        return Ok(std::u64::MAX);
+    }
+    let val = bytes.iter().fold(0, |acc, x| acc << 8 | (u64::from(*x)));
+    Ok(val)
+}
+
+impl BinaryLiteral {
+    /// from_u64 creates a new BinaryLiteral instance by the given uint value in BitEndian.
+    /// byte size will be used as the length of the new BinaryLiteral, with leading bytes filled to zero.
+    /// If byte size is -1, the leading zeros in new BinaryLiteral will be trimmed.
+    pub fn from_u64(val: u64, byte_size: isize) -> Result<Self> {
+        if byte_size != -1 && (byte_size < 1 || byte_size > 8) {
+            return Err(box_err!("invalid byte size: {}", byte_size));
+        }
+        let mut wtr = vec![];
```

Comment:
Maybe we could use: https://doc.rust-lang.org/std/primitive.u64.html#method.to_be_bytes